scripts/service-principals.sh

```diff
@@ -0,0 +1,40 @@
+set -ex
+
+CHECK_DAYS=$1
+
+TODAY_DATE=$(date +%Y-%m-%d)
+
+CHECK_DATE=$(date -d "+${CHECK_DAYS} days" +%Y-%m-%d)
+
+
+AZ_APP_RESULT=$( az ad app list --all --query "[?passwordCredentials[?endDateTime > '${TODAY_DATE}' && endDateTime < '${CHECK_DATE}']].{displayName:displayName, appId:appId, createdDateTime:createdDateTime, passwordCredentials:passwordCredentials[?endDateTime > '${TODAY_DATE}' && endDateTime < '${CHECK_DATE}'].{displayName:displayName,endDateTime:endDateTime}}" --output json )
+
+#AZ_APP_RESULT=$( az ad app list --all --query "[?passwordCredentials[?endDateTime > '2023-03-28' && endDateTime < '2023-04-29']].{displayName:displayName, appId:appId, createdDateTime:createdDateTime, passwordCredentials:passwordCredentials[?endDateTime > '2023-03-28' && endDateTime < '2023-04-29'].{displayName:displayName,endDateTime:endDateTime}}" --output json )
+
+AZ_APP_COUNT=$(jq -r '. | length' <<< "${AZ_APP_RESULT}")
+
+if [[ $AZ_APP_COUNT == 0 ]]; then
+    printf "\n:green_circle: None Service Principals Secrets are expiring in $CHECK_DAYS days \n\n" >> slack-message.txt
+    exit 0
+fi
+
+printf "\n:azure-826: <https://portal.azure.com/#view/Microsoft_AAD_IAM/ActiveDirectoryMenuBlade/~/RegisteredApps|_*Service Principal Secrets Status*_> \n" >> slack-message.txt
```

Comment:
this should probably move above line 16 , with current code  heading will come after green message .